airflow/operators/subdag.py

```diff
@@ -112,7 +113,7 @@ def _validate_pool(self, session):
             conflicts = [t for t in self.subdag.tasks if t.pool == self.pool]
             if conflicts:
                 # only query for pool conflicts if one may exist
-                pool = session.query(Pool).filter(Pool.slots == 1).filter(Pool.pool == self.pool).first()
+                pool = session.scalar(select(Pool).where(Pool.slots == 1, Pool.pool == self.pool))
```

Comment:
Also here, should we apply a limit to get the first item

Reply:
`Pool.pool` is also unique